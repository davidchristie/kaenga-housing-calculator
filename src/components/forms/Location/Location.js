import React, { Component, PropTypes } from 'react'
import { Button, Panel } from 'react-bootstrap'
import { Field } from 'redux-form'

import Select from '../../controls/Select'

class Location extends Component {
  render () {
    return (
      <form onSubmit={this.props.handleSubmit}>
        <Panel header='Location'>
          <h4>City</h4>
          <Button active>Auckland</Button>
          <Button disabled>Wellington</Button>
          <Button disabled>Christchurch</Button>
          <h4>Suburbs</h4>
          <Field
            component={Select}
            multiple
            name='suburbs'
          >
            {this.props.suburbs.map((suburb, index) => {
              return (
                <option key={index} value={suburb.name}>
                  {suburb.name}
                </option>
              )
            })}
          </Field>
        </Panel>
      </form>
    )
  }
}

Location.defaultProps = {
  suburbs: []
}

Location.propTypes = {
  handleSubmit: PropTypes.func,
  suburbs: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired
    })
  ).isRequired
}

export default Location
