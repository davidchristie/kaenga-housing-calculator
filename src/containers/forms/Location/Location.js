import { connect } from 'react-redux'
import { reduxForm } from 'redux-form'

import Location from '../../../components/forms/Location'
import { getSuburbs } from '../../../selectors/suburbs'

export default connect(
  state => {
    return {
      suburbs: getSuburbs(state)
    }
  }
)(
  reduxForm({
    destroyOnUnmount: false,
    form: 'location',
    initialValues: {
      commute: [],
      region: [],
      suburbs: []
    }
  })(Location)
)
