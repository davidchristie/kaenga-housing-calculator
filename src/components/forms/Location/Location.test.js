import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import Location from './Location'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<Location />)).toMatchSnapshot()
})
