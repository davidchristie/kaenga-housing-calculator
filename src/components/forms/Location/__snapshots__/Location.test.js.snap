exports[`test matches snapshot 1`] = `
<form
  onSubmit={undefined}>
  <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}
    header="Location">
    <h4>
      City
    </h4>
    <Button
      active={true}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}>
      Auckland
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={true}>
      Wellington
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={true}>
      Christchurch
    </Button>
    <h4>
      Suburbs
    </h4>
    <Field
      component={[Function]}
      multiple={true}
      name="suburbs">
      
    </Field>
  </Panel>
</form>
`;
